mithril-client/src/cardano_database_client/download_unpack.rs

```diff
@@ -0,0 +1,1204 @@
+            match result? {
+                Ok(immutable_file_number) => {
+                    immutable_file_numbers_downloaded.insert(immutable_file_number);
+                }
+                Err(e) => {
+                    slog::error!(
+                        self.logger,
+                        "Failed downloading and unpacking immutable files"; "error" => e.to_string(), "target_dir" => immutable_files_target_dir.display()
+                    );
+                }
+            }
+        }
+
+        Ok(immutable_file_numbers_downloaded)
+    }
+
+    fn spawn_immutable_download_future(
+        &self,
+        args: DownloadImmutableFutureBuilderArgs,
+    ) -> MithrilResult<Pin<Box<DownloadImmutableFuture>>> {
+        let feedback_receiver_clone = self.feedback_sender.clone();
+        let logger_clone = self.logger.clone();
+        let download_id_clone = args.download_id.to_string();
+        let file_downloader = args.file_downloader;
+        let file_downloader_uri = args.file_downloader_uri;
+        let compression_algorithm = args.compression_algorithm;
+        let immutable_files_target_dir = args.immutable_files_target_dir;
+        let immutable_file_number = args.immutable_file_number;
+        let file_size = args.file_size;
+        let download_future = async move {
+            feedback_receiver_clone
+                .send_event(MithrilEvent::CardanoDatabase(
+                    MithrilEventCardanoDatabase::ImmutableDownloadStarted {
+                        immutable_file_number,
+                        download_id: download_id_clone.clone(),
+                        size: file_size,
+                    },
+                ))
+                .await;
+            let downloaded = file_downloader
+                .download_unpack(
+                    &file_downloader_uri,
+                    &immutable_files_target_dir,
+                    Some(compression_algorithm),
+                    DownloadEvent::Immutable {
+                        immutable_file_number,
+                        download_id: download_id_clone.clone(),
+                    },
+                )
+                .await;
+            match downloaded {
+                Ok(_) => {
+                    feedback_receiver_clone
+                        .send_event(MithrilEvent::CardanoDatabase(
+                            MithrilEventCardanoDatabase::ImmutableDownloadCompleted {
+                                immutable_file_number,
+                                download_id: download_id_clone,
+                            },
+                        ))
+                        .await;
+
+                    Ok(immutable_file_number)
+                }
+                Err(e) => {
+                    slog::error!(
+                        logger_clone,
+                        "Failed downloading and unpacking immutable file {immutable_file_number} for location {file_downloader_uri:?}"; "error" => e.to_string()
```

Comment:
Should we debug the error to show the stack trace ? Else only the main message will be shown.
```suggestion
                        "Failed downloading and unpacking immutable file {immutable_file_number} for location {file_downloader_uri:?}"; "error" => ?e
```